### PR TITLE
Issue 1445 vaultdefaultlease

### PR DESCRIPTION
**Pull request by nritholtz**
Add support for vault default_lease_duration

Addresses #1445

Wanted to get the conversation going, so I am definitely OK with making changes.

### CODE REVIEW

**Comment by vfoucault** [comment]
Definitively what I was looking for !
In time, it will be a big deal to wait `5 * time.Second` before each path discovery ! Can't wait for this to be released.

GG.

**Comment by eikenb** [comment]
Thanks for the PR @nritholtz and for chiming in @vfoucault!

We've been focused on the development of a new project and are preparing for it's initial release. Once out we'll have more time for our other projects again and we'll get these PRs reviewed.

File: manager/runner.go [hunk 1]
```diff
@@ -852,6 +852,9 @@ func (r *Runner) init() error {
 	}
 	log.Printf("[DEBUG] (runner) final config: %s", result)
 
+	//Set VaultDefaultLeaseDuration
+	dep.VaultDefaultLeaseDuration = config.TimeDurationVal(r.config.Vault.DefaultLeaseDuration)
```
Comment (eikenb) [inline manager/runner.go:856]: Not a huge fan of using a module/global variable for this but there is no other good place for now. That is I'm considering this OK for the merge but will probably want to revisit this later... maybe when I port it to hashicat. 

Consider this more an internal note to myself than anything that needs to be changed here.

File: README.md [hunk 2]
```diff
@@ -418,6 +418,12 @@ vault {
   # before being used. Please see Vault's cubbyhole response wrapping
   # documentation for more information.
   unwrap_token = true
+  
+  # The default lease duration Consul Template will use on a Vault secret that 
+  # does not have a lease duration. This is used to calculate the sleep duration
+  # for rechecking a Vault secret value. This field is optional and will default to
+  # 5 minutes.
+  # default_lease_duration = "60s"
```
Comment (eikenb) [inline README.md:426]: This documentation has moved while this has been waiting (sorry). If you have a chance and wouldn't mind @nritholtz, please consider updating this PR with the documentation moved into `docs/configuration.md`. If you don't have time just say so, I can make the update if necessary.

Thanks.

Comment (eikenb) [reply README.md:426]: Ok, will do. I'll try to preserve attribution. Thanks.

**Comment by eikenb** [comment]
I had to do the PR locally and worked on it there such that github doesn't recognize it as this PR.

Closing with `manually merged`.

**Pull request closed without merging**